semantic-core/src/Analysis/Typecheck.hs

```diff
@@ -96,14 +96,14 @@ generalize ty = fromJust (closed (forAlls (IntSet.toList (mvs ty)) (hoistTerm R
 typecheckingFlowInsensitive
   :: Ord term
   => (forall sig m
-     .  (Carrier sig m, Member (Reader Path) sig, Member (Reader Span) sig, MonadFail m)
+     .  (Carrier sig m, Member (Reader Path.AbsRelFile) sig, Member (Reader Span) sig, MonadFail m)
      => Analysis term Name Type m
      -> (term -> m Type)
      -> (term -> m Type)
      )
   -> [File term]
   -> ( Heap Name Type
-     , [File (Either (Path, Span, String) (Term (Polytype :+: Monotype) Void))]
+     , [File (Either (Path.AbsRelFile, Span, String) (Term (Polytype :+: Monotype) Void))]
```

Comment:
Might be worth abstracting this triple into a real type.

Reply:
I had briefly used `File` for this, but they really don’t mean the same thing (the `Span` in a `File` should be the span of the _entire_ thing, whereas this span is expected to be localized). For the time being, I’d prefer to leave it as a triple until we have another use case for the localized context.